tools/common_lib/src/gemm.h

```diff
@@ -944,15 +950,33 @@ class GemmUmdD3d12Dispatcher : public GemmBaseDispatcher
     {
         using namespace dnnl_utils;
 
-        input_a_memory_desc_ = to_dnnl_mem_desc(params_.shape_a, params_.layout, params_.dt);
-        const auto input_b_memory_desc = to_dnnl_mem_desc(params_.shape_b, params_.b_managed ? DataLayout::eWeightsLayoutStart : params_.layout, params_.dt);
+        //input_a_memory_desc_ = to_dnnl_mem_desc(params_.a_transposed ? TensorShape{ params_.shape_a.n, params_.shape_a.c, params_.shape_a.w, params_.shape_a.h } : params_.shape_a, params_.layout, params_.dt);
+        if (params_.a_transposed)
+        {
+            input_a_memory_desc_ = to_dnnl_mem_desc(params_.shape_a, params_.layout, params_.dt);
+            input_a_memory_desc_ = matrix_transpose(input_a_memory_desc_, params_.dt);
+        }
+        else
+        {
+            input_a_memory_desc_ = to_dnnl_mem_desc(params_.shape_a, params_.layout, params_.dt);
+        }
+        // const auto input_b_memory_desc = to_dnnl_mem_desc(params_.b_transposed ? TensorShape{ params_.shape_b.n, params_.shape_b.c, params_.shape_b.w, params_.shape_b.h } : params_.shape_b, params_.b_managed ? DataLayout::eWeightsLayoutStart : params_.layout, params_.dt);
+        if (params_.b_transposed)
+        {
+            input_b_memory_desc = to_dnnl_mem_desc(params_.shape_b, params_.b_managed ? DataLayout::eWeightsLayoutStart : params_.layout, params_.dt);
+            input_b_memory_desc = matrix_transpose(input_b_memory_desc, params_.dt);
```

Comment:
If tensor is managed then we dont want to force any format tag and we want to leave format::any.
Using `matrix_transpose(...)` function here will force concrete layout, thus breaking managed functionality